packages/plugin-types/src/backend_secret_resolver.ts

```diff
@@ -14,10 +14,16 @@ export type BackendSecret = {
   /**
    * Resolves the given secret to its path
    */
-  resolvePath: (backendIdentifier: BackendIdentifier) => Record<string, string>;
+  resolvePath: (backendIdentifier: BackendIdentifier) => {
+    branchSecretPath: string;
+    sharedSecretPath: string;
+  };
```

Comment:
We can make this an exported type like `ResolvePathResponse` so we don't have to define this same type all over the place

Reply:
Yeah it does not complain with the function signature update so I have removed the if statement.

Also added the exported type `ResolvePathResponse`.